spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3PathMatchingResourcePatternResolver.java

```diff
@@ -0,0 +1,320 @@
+	 * case the resource should be modified and for the pattern matching a {@link AntPathMatcher} is used.
+	 *
+	 * @param s3Client the s3Client of the Amazon SDK
+	 * @param resourcePatternResolverDelegate the resourcePatternResolverDelegate which is used if the given scheme is
+	 *     not S3. In this case all processing is delegated to this implementation.
+	 */
+	public S3PathMatchingResourcePatternResolver(S3Client s3Client,
+			ResourcePatternResolver resourcePatternResolverDelegate) {
+		this(s3Client, resourcePatternResolverDelegate,
+				new InMemoryBufferingS3OutputStreamProvider(s3Client, new PropertiesS3ObjectContentTypeResolver()),
+				new AntPathMatcher(), List.of("**", "*", "?"));
+	}
+
+	/**
+	 * Creates a {@link S3PathMatchingResourcePatternResolver} with the given s3Client and
+	 * resourcePatternResolverDelegate.
+	 *
+	 * @param s3Client the s3Client of the Amazon SDK
+	 * @param resourcePatternResolverDelegate the resourcePatternResolverDelegate which is used if the given scheme is
+	 *     not S3. In this case all processing is delegated to this implementation.
+	 * @param s3OutputStreamProvider The s3OutputStreamProvider used if the resource is going to be written
+	 * @param pathMatcher used to resolve resources and bucket names
+	 * @param wildCardSeparators the wildcard separators to determine the prefix. You can also use
+	 *     Collections.emptyList(), but this will cause performance and request impacts.
+	 */
+	public S3PathMatchingResourcePatternResolver(S3Client s3Client,
+			ResourcePatternResolver resourcePatternResolverDelegate, S3OutputStreamProvider s3OutputStreamProvider,
+			PathMatcher pathMatcher, List<String> wildCardSeparators) {
+		Assert.notNull(s3Client, "S3Client must not be null");
+		Assert.notNull(resourcePatternResolverDelegate, "ResourcePatternResolver must not be null");
+		Assert.notNull(s3OutputStreamProvider, "S3OutputStreamProvider must not be null");
+		Assert.notNull(pathMatcher, "PathMatcher must not be null");
+		Assert.notNull(wildCardSeparators, "WildCardSeparators must not be null");
+		this.s3Client = s3Client;
+		this.resourcePatternResolverDelegate = resourcePatternResolverDelegate;
+		this.s3OutputStreamProvider = s3OutputStreamProvider;
+		this.pathMatcher = pathMatcher;
+		this.wildCardSeparators = wildCardSeparators;
+	}
+
+	/**
+	 * Gets all resources based on the given location pattern. If the location pattern is not using the s3 scheme the
+	 * operation is delegated to resourcePatternResolverDelegate.
+	 *
+	 * @param locationPattern the location pattern to get all resources from
+	 * @return an array with all resources
+	 * @throws IOException if something went wrong during the resource resolving
+	 */
+	@Override
+	public Resource[] getResources(String locationPattern) throws IOException {
+		LOGGER.debug("Get resources of the following location pattern {}", locationPattern);
```

Comment:
Actually now thinking about it maybe it can be null since some `resourcePatternResolverDelegate` can maybe resolve it.
But our code will fail with a null pointer before this is exectued.
@maciejwalkowiak

Reply:
The interface does not tell if it can be null, but implementations like Spring's `PathMatchingResourcePatternResolver` also assert the location pattern to be not null. For the return value the stream is fine, because if the resource array is empty the stream does nothing an empty array will be returned.